### ISSUE TITLE
com.google.api.client.http.HttpRequest.execute doesn't close Span

### ISSUE THREAD

**Issue by abdolence** [googleapis/google-http-java-client#788]
#### Environment details
1. Core/Tracing API
2. OS type and version: Linux
3. Java version: 1.8
4. google-http-client version(s): 1.30.1

#### Steps to reproduce

  1. Just called `ServiceOptions.getDefaultProjectId()` with opencensus Tracer enabled (I'm using Stackdriver). It apparently creates a span in `com.google.api.client.http.HttpRequest.execute()`, and throws inner exception `(UnknownHostException : metadata.google.internal) and forgetting to close span at this moment in the exception handler.
For my version of google http client this place:

```java
catch (IOException e) {
        if (!retryOnExecuteIOException
            && (ioExceptionHandler == null
                || !ioExceptionHandler.handleIOException(this, retryRequest))) {
          throw e;
        }
        // Save the exception in case the retries do not work and we need to re-throw it later.
        executeException = e;
        if (loggable) {
          logger.log(Level.WARNING, "exception thrown while executing request", e);
        }
      } finally {
        ws.close();
      }
```

  2. It also will be found by opencensus finalize afterwards:
```
Aug 10, 2019 7:45:59 PM io.opencensus.implcore.trace.RecordEventsSpanImpl finalize
SEVERE: Span Sent.com.google.api.client.http.HttpRequest.execute is GC'ed without being ended.
```

#### Code example

```java
ServiceOptions.getDefaultProjectId()
```

**Comment by songy23** [comment]
Also reported in https://github.com/census-instrumentation/opencensus-java/issues/1963.

**Comment by chingor13** [comment]
Hmm, the span should be closed because we close the span via the scope in the finally section of [a try/catch/finally block](https://github.com/googleapis/google-http-java-client/blob/04d27047d5e0e1860eff86403005535f8f7ec079/google-http-client/src/main/java/com/google/api/client/http/HttpRequest.java#L992-L1025).

**Comment by chingor13** [comment]
Ah, closing the scope, doesn't end the span.

**Comment by abdolence** [comment]
> Ah, closing the scope, doesn't end the span.

That's right. If you were expecting automatic `Span.end()` then it was working only with `Tracer.spanBuilder....startScopedSpan()` instead `Tracer.withSpan()` (correct me if I'm wrong).
